lib/views/new_todo_page.dart

```diff
@@ -1,20 +1,43 @@
 import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
 
-class NewTodoPage extends StatelessWidget {
+import '../cubit/todolist_cubit.dart';
+import '../models/todo.dart';
+
+class NewTodoPage extends StatefulWidget {
   const NewTodoPage({Key? key}) : super(key: key);
 
+  @override
+  State<NewTodoPage> createState() => _NewTodoPageState();
+}
+
+class _NewTodoPageState extends State<NewTodoPage> {
+  TextEditingController tittleController = TextEditingController();
+
+  TextEditingController descriptionController = TextEditingController();
+
   @override
   Widget build(BuildContext context) {
     return Scaffold(
       appBar: AppBar(
         actions: [
           TextButton(
-              style: TextButton.styleFrom(
-                  textStyle: const TextStyle(
-                      fontSize: 18, fontWeight: FontWeight.w400),
-                  primary: Colors.white),
-              onPressed: () => {Navigator.pop(context)},
-              child: const Text('Save'))
+            style: TextButton.styleFrom(
+                textStyle:
+                    const TextStyle(fontSize: 18, fontWeight: FontWeight.w400),
+                primary: Colors.white),
+            onPressed: () {
+              Todo item = Todo(
+                  title: tittleController.text,
+                  description: descriptionController.text);
+              BlocProvider.of<TodolistCubit>(context).addTodo(item);
```

Comment:
what about an extension with a generic type

Reply:
Like this? 
```suggestion
   extension ArgumentExtractor on BuildContext {
         Object? get arguments => ModalRoute.of(this)!.settings.arguments;
         TodolistCubit get todoListCubit => BlocProvider.of<TodolistCubit>(this);
}
```

Reply:
think of a way to do it more general for all cubits or blocs you want to use